regulation-worker/cmd/main.go

```diff
@@ -79,13 +85,18 @@ func Run(ctx context.Context) error {
 	backendconfig.DefaultBackendConfig.WaitForConfig(ctx)
 	identity := backendconfig.DefaultBackendConfig.Identity()
 	dest.Start(ctx)
-
+	oauthV2Enabled := config.GetReloadableBoolVar(false, "RegulationWorker.oauthV2Enabled")
+	// TODO: Remove later
+	pkgLogger.Infon("[regulationApi]", logger.NewBoolField("oauthV2Enabled", oauthV2Enabled.Load()))
+	httpTimeout := config.GetReloadableDurationVar(60, time.Second, "HttpClient.regulationWorker.regulationManager.timeout")
 	// setting up oauth
 	OAuth := oauth.NewOAuthErrorHandler(backendconfig.DefaultBackendConfig, oauth.WithRudderFlow(oauth.RudderFlow_Delete))
 
+	cli := createHTTPClient(config.Default, httpTimeout, oauthV2Enabled)
+
 	svc := service.JobSvc{
 		API: &client.JobAPI{
-			Client:    &http.Client{Timeout: config.GetDuration("HttpClient.regulationWorker.regulationManager.timeout", 60, time.Second)},
+			Client:    &http.Client{Timeout: httpTimeout.Load()},
```

Comment:
`cli` is http client that is used to connect to transformer.
`Client` in `client.JobAPI` is used to connect to config-backend to get information about available regulation jobs.
Since they are fundamentally serving different use-cases, we let that be.

Also cli now can contain interceptor depending on the destination, as such we don't need interceptor at all for an API call which just gets the available regulation jobs from config-be(I mean regulation-manager)

Reply:
Let me know if it makes sense